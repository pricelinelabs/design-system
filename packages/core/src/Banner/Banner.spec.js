import React from 'react'
import { render } from 'testing-library'

import { Banner, Text, Heading, theme } from '..'

describe('Banner', () => {
  let consoleError
  beforeEach(() => {
    consoleError = console.error
    console.error = jest.fn()
  })
  afterEach(() => (console.error = consoleError))

  test('renders with no props other than theme', () => {
    const json = rendererCreateWithTheme(<Banner />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with custom iconName and size', () => {
    const json = rendererCreateWithTheme(
      <Banner iconName='star' iconSize={20} />
    ).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with text string', () => {
    const json = rendererCreateWithTheme(
      <Banner
        header='Header'
        text='Text'
        iconName='star'
        iconSize={20}
        theme={theme}
      />
    ).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with text node', () => {
    const json = rendererCreateWithTheme(
      <Banner
        header='Header'
        text={<Text>Text</Text>}
        iconName='star'
        iconSize={20}
        theme={theme}
      />
    ).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with primary color', () => {
    const json = rendererCreateWithTheme(<Banner color='primary' />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.blue)
    expect(json).toHaveStyleRule('color', theme.colors.white)
  })

  test('renders with secondary color', () => {
    const json = rendererCreateWithTheme(<Banner color='secondary' />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.green)
    expect(json).toHaveStyleRule('color', theme.colors.white)
  })

  test('renders with alert color', () => {
    const json = rendererCreateWithTheme(<Banner color='alert' />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.orange)
    expect(json).toHaveStyleRule('color', theme.colors.white)
  })

  test('renders with error color', () => {
    const json = rendererCreateWithTheme(<Banner color='error' />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.red)
    expect(json).toHaveStyleRule('color', theme.colors.white)
  })

  test('renders with primary.light color', () => {
    const json = rendererCreateWithTheme(
      <Banner color='primary.light' />
    ).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.lightBlue)
    expect(json).toHaveStyleRule('color', theme.colors.darkBlue)
  })

  test('renders with secondary.light color', () => {
    const json = rendererCreateWithTheme(
      <Banner color='secondary.light' />
    ).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.lightGreen)
    expect(json).toHaveStyleRule('color', theme.colors.darkGreen)
  })

  test('renders with error.light color', () => {
    const json = rendererCreateWithTheme(
      <Banner color='error.light' />
    ).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.lightRed)
    expect(json).toHaveStyleRule('color', theme.colors.darkRed)
  })

  test('renders close button if onClose func is provided', () => {
    const { container } = render(<Banner onClose={() => {}} />)
    const closeButton = container.querySelector('button')
    expect(closeButton).toBeTruthy()
  })

  test('does render blue left-hand icon by default', () => {
    const { container } = render(<Banner color='primary' />)
    const icon = container.querySelector('svg')
    // todo: check svg path/attrs for correct icon
    expect(icon).toBeTruthy()
  })

  test('does not render blue left-hand icon if showIcon is false', () => {
    const { container } = render(<Banner color='primary' showIcon={false} />)
    const icon = container.querySelector('svg')
    expect(icon).toBeFalsy()
  })

  test('renders content from children props', () => {
    const json = rendererCreateWithTheme(
      <Banner>
        <span>123</span>
      </Banner>
    ).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with header node', () => {
    const { asFragment, getByText } = render(
      <Banner
        header={<Heading>Hello world</Heading>}
        text={<Text>Text</Text>}
        iconName='star'
        iconSize={20}
        theme={theme}
      />
    )

    getByText(/Hello world/)
    expect(asFragment()).toMatchSnapshot()
  })
})
