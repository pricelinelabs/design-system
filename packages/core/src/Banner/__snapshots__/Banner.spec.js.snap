// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner renders content from children props 1`] = `
.c0 {
  text-align: left;
  background-color: #fff;
  color: #001833;
}

.c5 {
  text-align: left;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  margin-top: -2px;
  margin-right: 8px;
  width: 24px;
}

.c4 {
  outline: none;
}

.c1 {
  background-color: #0a0;
  color: #fff;
}

<div
  className="c0 c1"
  color="text.lightest"
>
  <div
    className="c2"
  >
    <svg
      aria-hidden={true}
      className="c3 c4"
      fill="currentcolor"
      focusable={false}
      height={24}
      role="img"
      tabIndex={-1}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm-1.7 14.5L6 12.2 7.2 11l3.1 3.1 6.5-6.5L18 8.8l-7.7 7.7z"
      />
    </svg>
    <div
      className="c5"
      width={1}
    >
      <span
        className="c6"
        fontSize={1}
      />
      <span>
        123
      </span>
    </div>
  </div>
</div>
`;

exports[`Banner renders with alert color 1`] = `
.c0 {
  text-align: left;
  background-color: #fff;
  color: #001833;
}

.c5 {
  text-align: left;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  margin-top: -2px;
  margin-right: 8px;
  width: 24px;
}

.c4 {
  outline: none;
}

.c1 {
  background-color: #f68013;
  color: #fff;
}

<div
  className="c0 c1"
  color="text.lightest"
>
  <div
    className="c2"
  >
    <svg
      aria-hidden={true}
      className="c3 c4"
      fill="currentcolor"
      focusable={false}
      height={24}
      role="img"
      tabIndex={-1}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M1 21.5h22l-11-19-11 19zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
      />
    </svg>
    <div
      className="c5"
      width={1}
    >
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with custom iconName and size 1`] = `
.c0 {
  text-align: left;
  background-color: #fff;
  color: #001833;
}

.c5 {
  text-align: left;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  margin-top: -2px;
  margin-right: 8px;
  width: 24px;
}

.c4 {
  outline: none;
}

.c1 {
  background-color: #0a0;
  color: #fff;
}

<div
  className="c0 c1"
  color="text.lightest"
>
  <div
    className="c2"
  >
    <svg
      aria-hidden={true}
      className="c3 c4"
      fill="currentcolor"
      focusable={false}
      height={24}
      role="img"
      tabIndex={-1}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm-1.7 14.5L6 12.2 7.2 11l3.1 3.1 6.5-6.5L18 8.8l-7.7 7.7z"
      />
    </svg>
    <div
      className="c5"
      width={1}
    >
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with error color 1`] = `
.c0 {
  text-align: left;
  background-color: #fff;
  color: #001833;
}

.c5 {
  text-align: left;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  margin-top: -2px;
  margin-right: 8px;
  width: 24px;
}

.c4 {
  outline: none;
}

.c1 {
  background-color: #c00;
  color: #fff;
}

<div
  className="c0 c1"
  color="text.lightest"
>
  <div
    className="c2"
  >
    <svg
      aria-hidden={true}
      className="c3 c4"
      fill="currentcolor"
      focusable={false}
      height={24}
      role="img"
      tabIndex={-1}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm1 11h-2V7h2v6zm0 4h-2v-2h2v2z"
      />
    </svg>
    <div
      className="c5"
      width={1}
    >
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with error.light color 1`] = `
.c0 {
  text-align: left;
  background-color: #800;
  color: #fff;
}

.c5 {
  text-align: left;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  margin-top: -2px;
  margin-right: 8px;
  width: 24px;
}

.c4 {
  outline: none;
}

.c1 {
  background-color: #fbebeb;
  color: #800;
}

<div
  className="c0 c1"
  color="error.dark"
>
  <div
    className="c2"
  >
    <svg
      aria-hidden={true}
      className="c3 c4"
      fill="currentcolor"
      focusable={false}
      height={24}
      role="img"
      tabIndex={-1}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm1 11h-2V7h2v6zm0 4h-2v-2h2v2z"
      />
    </svg>
    <div
      className="c5"
      width={1}
    >
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with header node 1`] = `
<DocumentFragment>
  .c1 {
  text-align: left;
  color: text.lightest;
}

.c6 {
  text-align: left;
  width: 100%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c7 {
  font-size: 14px;
}

.c4 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  margin-top: -2px;
  margin-right: 8px;
  width: 24px;
}

.c5 {
  outline: none;
}

.c2 {
  background-color: secondary.base;
  color: text.lightest;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

<div
    class="c0"
  >
    <div
      class="c1 c2"
      color="text.lightest"
    >
      <div
        class="c3"
      >
        <svg
          aria-hidden="true"
          class="c4 c5"
          fill="currentcolor"
          focusable="false"
          height="24"
          role="img"
          tabindex="-1"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm-1.7 14.5L6 12.2 7.2 11l3.1 3.1 6.5-6.5L18 8.8l-7.7 7.7z"
          />
        </svg>
        <div
          class="c6"
          width="1"
        >
          <h3
            class=""
          >
            Hello world
          </h3>
          <span
            class="c7"
            font-size="1"
          >
            <div
              class=""
            >
              Text
            </div>
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Banner renders with no props other than theme 1`] = `
.c0 {
  text-align: left;
  background-color: #fff;
  color: #001833;
}

.c5 {
  text-align: left;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  margin-top: -2px;
  margin-right: 8px;
  width: 24px;
}

.c4 {
  outline: none;
}

.c1 {
  background-color: #0a0;
  color: #fff;
}

<div
  className="c0 c1"
  color="text.lightest"
>
  <div
    className="c2"
  >
    <svg
      aria-hidden={true}
      className="c3 c4"
      fill="currentcolor"
      focusable={false}
      height={24}
      role="img"
      tabIndex={-1}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm-1.7 14.5L6 12.2 7.2 11l3.1 3.1 6.5-6.5L18 8.8l-7.7 7.7z"
      />
    </svg>
    <div
      className="c5"
      width={1}
    >
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with primary color 1`] = `
.c0 {
  text-align: left;
  background-color: #fff;
  color: #001833;
}

.c5 {
  text-align: left;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  margin-top: -2px;
  margin-right: 8px;
  width: 24px;
}

.c4 {
  outline: none;
}

.c1 {
  background-color: #0068EF;
  color: #fff;
}

<div
  className="c0 c1"
  color="text.lightest"
>
  <div
    className="c2"
  >
    <svg
      aria-hidden={true}
      className="c3 c4"
      fill="currentcolor"
      focusable={false}
      height={24}
      role="img"
      tabIndex={-1}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
      />
    </svg>
    <div
      className="c5"
      width={1}
    >
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with primary.light color 1`] = `
.c0 {
  text-align: left;
  background-color: #049;
  color: #fff;
}

.c5 {
  text-align: left;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  margin-top: -2px;
  margin-right: 8px;
  width: 24px;
}

.c4 {
  outline: none;
}

.c1 {
  background-color: #e8f2ff;
  color: #049;
}

<div
  className="c0 c1"
  color="primary.dark"
>
  <div
    className="c2"
  >
    <svg
      aria-hidden={true}
      className="c3 c4"
      fill="currentcolor"
      focusable={false}
      height={24}
      role="img"
      tabIndex={-1}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
      />
    </svg>
    <div
      className="c5"
      width={1}
    >
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with secondary color 1`] = `
.c0 {
  text-align: left;
  background-color: #fff;
  color: #001833;
}

.c5 {
  text-align: left;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  margin-top: -2px;
  margin-right: 8px;
  width: 24px;
}

.c4 {
  outline: none;
}

.c1 {
  background-color: #0a0;
  color: #fff;
}

<div
  className="c0 c1"
  color="text.lightest"
>
  <div
    className="c2"
  >
    <svg
      aria-hidden={true}
      className="c3 c4"
      fill="currentcolor"
      focusable={false}
      height={24}
      role="img"
      tabIndex={-1}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm-1.7 14.5L6 12.2 7.2 11l3.1 3.1 6.5-6.5L18 8.8l-7.7 7.7z"
      />
    </svg>
    <div
      className="c5"
      width={1}
    >
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with secondary.light color 1`] = `
.c0 {
  text-align: left;
  background-color: #060;
  color: #fff;
}

.c5 {
  text-align: left;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  margin-top: -2px;
  margin-right: 8px;
  width: 24px;
}

.c4 {
  outline: none;
}

.c1 {
  background-color: #ecf7ec;
  color: #060;
}

<div
  className="c0 c1"
  color="secondary.dark"
>
  <div
    className="c2"
  >
    <svg
      aria-hidden={true}
      className="c3 c4"
      fill="currentcolor"
      focusable={false}
      height={24}
      role="img"
      tabIndex={-1}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm-1.7 14.5L6 12.2 7.2 11l3.1 3.1 6.5-6.5L18 8.8l-7.7 7.7z"
      />
    </svg>
    <div
      className="c5"
      width={1}
    >
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with text node 1`] = `
.c0 {
  text-align: left;
  color: text.lightest;
}

.c5 {
  text-align: left;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 16px;
  font-weight: 700;
  line-height: 1.25;
}

.c7 {
  font-size: 14px;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  margin-top: -2px;
  margin-right: 8px;
  width: 24px;
}

.c4 {
  outline: none;
}

.c1 {
  background-color: secondary.base;
  color: text.lightest;
}

<div
  className="c0 c1"
  color="text.lightest"
>
  <div
    className="c2"
  >
    <svg
      aria-hidden={true}
      className="c3 c4"
      fill="currentcolor"
      focusable={false}
      height={24}
      role="img"
      tabIndex={-1}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm-1.7 14.5L6 12.2 7.2 11l3.1 3.1 6.5-6.5L18 8.8l-7.7 7.7z"
      />
    </svg>
    <div
      className="c5"
      width={1}
    >
      <div
        className="c6"
      >
        Header
      </div>
      <span
        className="c7"
        fontSize={1}
      >
        <div
          className=""
        >
          Text
        </div>
      </span>
    </div>
  </div>
</div>
`;

exports[`Banner renders with text string 1`] = `
.c0 {
  text-align: left;
  color: text.lightest;
}

.c5 {
  text-align: left;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 16px;
  font-weight: 700;
  line-height: 1.25;
}

.c7 {
  font-size: 14px;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  margin-top: -2px;
  margin-right: 8px;
  width: 24px;
}

.c4 {
  outline: none;
}

.c1 {
  background-color: secondary.base;
  color: text.lightest;
}

<div
  className="c0 c1"
  color="text.lightest"
>
  <div
    className="c2"
  >
    <svg
      aria-hidden={true}
      className="c3 c4"
      fill="currentcolor"
      focusable={false}
      height={24}
      role="img"
      tabIndex={-1}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm-1.7 14.5L6 12.2 7.2 11l3.1 3.1 6.5-6.5L18 8.8l-7.7 7.7z"
      />
    </svg>
    <div
      className="c5"
      width={1}
    >
      <div
        className="c6"
      >
        Header
      </div>
      <span
        className="c7"
        fontSize={1}
      >
        Text
      </span>
    </div>
  </div>
</div>
`;
